independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientResponseCompleteRestHandler.java

```diff
@@ -22,7 +23,10 @@ public static ResponseImpl mapToResponse(RestClientRequestContext context, boole
         builder.status(responseContext.getStatus(), responseContext.getReasonPhrase());
         builder.setAllHeaders(responseContext.getHeaders());
         builder.invocationState(context);
-        if (context.isResponseTypeSpecified() && parseContent) { // this case means that a specific response type was requested
+        if (context.isResponseTypeSpecified()
+                // when we are returning a RestResponse, we don't want to do any parsing
+                && (Response.Status.Family.familyOf(context.getResponseStatus()) == Response.Status.Family.SUCCESSFUL)
```

Comment:
was the check for status code missing before or was it done later in the process?

Reply:
It was also being done in `ClientSetResponseEntityRestHandler` which is called before this.
We also need here now, due to the use of `RestResponse`. With `Response` it wasn't an issue because the type was not set.